src/common/m_pc_buffer.cpp

```diff
@@ -43,37 +44,77 @@ bool PCBuffer::get(Block* &block, int column) {
 	return true;
 }
 
-//bool PCBuffer::get_sorted(Block *&block) {
-//	if(!begin_) {
-//		for(int i=0; i<row_; i++) {
-//			merged_blocks_[i]->storeBlock(data_[i]->pop()->getAddr(), BLOCK_SIZE);
-//			itrs_[i]=merged_blocks_[i]->createIterator();
-//		}
-//		begin_=true;
-//	}
-//
-//	void *desc=0;
-//	while((desc=block->allocateTuple())!=0) {
-//		void *tuple=0;
-//		void *most=0;
-//		int most_cur=0;
-//		for(int i=0; i<row_; i++) {
-//			if(!itrs_[i]->getCurrent()==0) {
-//				if(data_[i]->pop()->get_size()==0) {
-//
-//				}
-//			}
-//		}
-//		/* choose the needed one tuple. */
-//		block->storeTuple(desc, tuple);
-//	}
-//
-//	/* build a 0-size one block. */
-//	return true;
-//}
+bool maxLast(void *start, NewSchema *schema) {
+	char *p=(char *)malloc(schema->get_bytes());
+	for(int i=0;i<schema->vd_.size();i++) {
+		if(schema->vd_[i]->get_type()==t_long) {
+			*(unsigned long *)(p+8*i)=ULONG_MAX;
+		}
+	}
+	start=p;
+}
+
+bool PCBuffer::get_sorted(Block *&block) {
```

Comment:
this is a globally sort function in the pcbuffer, we can get the globally sorted data by using this function, it's by loser tree, which can be a good way to merge the sorted data.